bpf/lib/nodeport.h

```diff
@@ -298,11 +298,12 @@ static __always_inline int dsr_set_ext6(struct __ctx_buff *ctx,
 {
 	struct dsr_opt_v6 opt __align_stack_8 = {};
 	__u16 payload_len = bpf_ntohs(ip6->payload_len) + sizeof(opt);
+	__u16 total_len = bpf_ntohs(ip6->payload_len) + sizeof(struct ipv6hdr) + sizeof(opt);
 
 	/* The IPv6 extension should be 8-bytes aligned */
 	build_bug_on((sizeof(struct dsr_opt_v6) % 8) != 0);
 
-	if (dsr_is_too_big(ctx, payload_len)) {
+	if (dsr_is_too_big(ctx, total_len)) {
```

Comment:
You've changed the way that we determine whether the packet is too big. I think that makes sense but I think we should have a bug report for this as it looks like a bug no?

Even if we don't want to report a bug. You changed the way we compute size for ipv6 but not for ipv4, any reason that I may have missed?

Reply:
I added the comment about this change elsewhere. 
https://github.com/nnbu/cilium/commit/ea404982e00fbd835c07f3b5f235a0e869b8094a#r84022786